SolastaCommunityExpansion/Api/BaseDefinitionBuilder.cs

```
@@ -31,6 +33,61 @@ protected static void LogDefinition(string msg)
                 Main.Log(msg);
             }
         }
+
+        protected static void VerifyDefinitionNameIsNotInUse(string name)
+        {
+            if (DisableVerifyDefinitionNameIsNotInUse)
+            {
+                return;
+            }
+
+            // Verify name has not been used previously
+            // 1) get all names used in all TA databases (at this point) ignoring existing duplicates 
+            // 2) check 'name' hasn't been used already, but ignore names we know already have duplicates
+
+            if (KnownDuplicateDefinitionNames.Contains(name))
+            {
+                return;
+            }
+
+            if (definitionNames.ContainsKey(name))
+            {
+                var definitionInDb = definitionNames[name];
+
+                var msg = definitionInDb ? "Non-CE definition:" : "CE definition";
+
+                throw new SolastaModApiException($"{msg}:{name} is already in use.");
```

Should we throw the error in release builds or only debug?

On the one hand the error could help identify conflicts that cause issues, but it also could mean breaking CE because another mod introduces a name conflict that doesn't actually conflict (different dbs).

---

I've improved the exception message.
Now throws in debug and logs in release.